src/data/java_profile.rs

```diff
@@ -0,0 +1,340 @@
+extern crate ctor;
+
+use crate::data::{CollectData, CollectorParams, Data, DataType, ProcessedData};
+use crate::visualizer::{DataVisualizer, GetData};
+use crate::{PERFORMANCE_DATA, VISUALIZATION_DATA};
+use anyhow::Result;
+use ctor::ctor;
+use log::{debug, error, trace};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::io::{ErrorKind, Write};
+use std::path::PathBuf;
+use std::process::{Child, Command};
+use std::sync::Mutex;
+use std::{fs, fs::File};
+
+pub static JAVA_PROFILE_FILE_NAME: &str = "java_profile";
+
+lazy_static! {
+    pub static ref ASPROF_CHILDREN: Mutex<Vec<Child>> = Mutex::new(Vec::new());
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct JavaProfileRaw {
+    process_map: HashMap<String, Vec<String>>,
+}
+
+impl JavaProfileRaw {
+    pub fn new() -> Self {
+        JavaProfileRaw {
+            process_map: HashMap::new(),
+        }
+    }
+
+    fn launch_asprof(&self, jids: Vec<String>, params: CollectorParams) -> Result<()> {
+        let data_dir = PathBuf::from(params.data_dir.clone());
+        for jid in &jids {
```

Comment:
jid is simply the pid of a (presumably) java process, right?

Reply:
Yes